### ISSUE TITLE
Incorrect index ranges due to index retention shortly after deflector cycling

### ISSUE THREAD

**Issue by JeremyEinfeld** [Graylog2/graylog2-server#1132]
The index range calculations following retention are not run following index retention if the index is cycled shortly before the index retention thread runs ("Maximum concurrency of job is reached" when it tries to run the index calculation job). This can result in incorrect index range data, with an index existing in index_ranges but not elasticsearch.

Index graylog2_2655 is deleted, but still exists in the range calculations following deletion:

```
2015-04-26 20:36:01,909 INFO : org.graylog2.periodical.IndexRotationThread - Deflector index <graylog2_2674> should be rotated, Pointing deflector to new index now!
2015-04-26 20:36:01,909 INFO : org.graylog2.indexer.Deflector - Cycling deflector to next index now.
2015-04-26 20:36:02,092 INFO : org.graylog2.indexer.Deflector - Cycling from <graylog2_2674> to <graylog2_2675>
2015-04-26 20:36:02,092 INFO : org.graylog2.indexer.Deflector - Creating index target <graylog2_2675>...
2015-04-26 20:36:06,542 INFO : org.graylog2.indexer.Deflector - Done!
2015-04-26 20:36:06,542 INFO : org.graylog2.indexer.Deflector - Pointing deflector to new target index....
2015-04-26 20:36:06,646 INFO : org.graylog2.system.jobs.SystemJobManager - Submitted SystemJob <dd414340-ec53-11e4-9fc2-fa163e8ba044> [org.graylog2.indexer.SetIndexReadOnlyJob]
2015-04-26 20:36:06,647 INFO : org.graylog2.system.jobs.SystemJobManager - Submitted SystemJob <dd41b870-ec53-11e4-9fc2-fa163e8ba044> [org.graylog2.indexer.ranges.RebuildIndexRangesJob]
2015-04-26 20:36:06,647 INFO : org.graylog2.indexer.Deflector - Done!
2015-04-26 20:36:06,648 INFO : org.graylog2.indexer.ranges.RebuildIndexRangesJob - Re-calculating index ranges.
2015-04-26 20:36:08,194 INFO : org.graylog2.indexer.ranges.RebuildIndexRangesJob - Calculated range of [graylog2_2666] in [1123ms].
2015-04-26 20:36:08,801 INFO : org.graylog2.indexer.ranges.RebuildIndexRangesJob - Calculated range of [graylog2_2667] in [606ms].
2015-04-26 20:36:11,103 INFO : org.graylog2.indexer.ranges.RebuildIndexRangesJob - Calculated range of [graylog2_2655] in [2301ms].
2015-04-26 20:36:11,234 INFO : org.graylog2.indexer.ranges.RebuildIndexRangesJob - Calculated range of [graylog2_2664] in [130ms].
2015-04-26 20:36:12,505 INFO : org.graylog2.indexer.ranges.RebuildIndexRangesJob - Calculated range of [graylog2_2656] in [1270ms].
2015-04-26 20:36:12,569 INFO : org.graylog2.indexer.ranges.RebuildIndexRangesJob - Calculated range of [graylog2_2665] in [64ms].
2015-04-26 20:36:13,026 INFO : org.graylog2.indexer.ranges.RebuildIndexRangesJob - Calculated range of [graylog2_2675] in [456ms].
2015-04-26 20:36:13,984 INFO : org.graylog2.indexer.ranges.RebuildIndexRangesJob - Calculated range of [graylog2_2657] in [957ms].
2015-04-26 20:36:15,111 INFO : org.graylog2.indexer.ranges.RebuildIndexRangesJob - Calculated range of [graylog2_2658] in [1127ms].
2015-04-26 20:36:16,217 INFO : org.graylog2.indexer.ranges.RebuildIndexRangesJob - Calculated range of [graylog2_2668] in [1105ms].
2015-04-26 20:36:18,040 INFO : org.graylog2.indexer.ranges.RebuildIndexRangesJob - Calculated range of [graylog2_2659] in [1823ms].
2015-04-26 20:36:24,481 INFO : org.graylog2.indexer.ranges.RebuildIndexRangesJob - Calculated range of [graylog2_2669] in [6439ms].
2015-04-26 20:36:25,486 INFO : org.graylog2.indexer.ranges.RebuildIndexRangesJob - Calculated range of [graylog2_2672] in [1005ms].
2015-04-26 20:36:26,173 INFO : org.graylog2.indexer.ranges.RebuildIndexRangesJob - Calculated range of [graylog2_2671] in [686ms].
2015-04-26 20:36:36,646 INFO : org.graylog2.indexer.SetIndexReadOnlyJob - Flushing old index <graylog2_2674>.
2015-04-26 20:36:44,038 INFO : org.graylog2.indexer.SetIndexReadOnlyJob - Setting old index <graylog2_2674> to read-only.
2015-04-26 20:36:44,181 INFO : org.graylog2.system.jobs.SystemJobManager - Submitted SystemJob <f3a0cd41-ec53-11e4-9fc2-fa163e8ba044> [org.graylog2.indexer.indices.jobs.OptimizeIndexJob]
2015-04-26 20:36:44,182 INFO : org.graylog2.system.jobs.SystemJobManager - SystemJob <dd414340-ec53-11e4-9fc2-fa163e8ba044> [org.graylog2.indexer.SetIndexReadOnlyJob] finished in 7535ms.
2015-04-26 20:36:44,189 INFO : org.graylog2.indexer.indices.jobs.OptimizeIndexJob - Optimizing index <graylog2_2674>.
2015-04-26 20:36:44,218 INFO : org.graylog2.indexer.ranges.RebuildIndexRangesJob - Calculated range of [graylog2_2674] in [18044ms].
2015-04-26 20:36:50,900 INFO : org.graylog2.periodical.IndexRetentionThread - Number of indices (21) higher than limit (20). Running retention for 1 indices.
2015-04-26 20:36:50,946 INFO : org.graylog2.periodical.IndexRetentionThread - Running retention strategy [org.graylog2.indexer.retention.strategies.DeletionRetentionStrategy] for index <graylog2_2655>
2015-04-26 20:36:50,979 INFO : org.graylog2.plugin.indexer.retention.RetentionStrategy - Strategy is deleting.
2015-04-26 20:36:51,065 INFO : org.graylog2.plugin.indexer.retention.RetentionStrategy - Finished index retention strategy [org.graylog2.indexer.retention.strategies.DeletionRetentionStrategy] for index <graylog2_2655> in 117ms.
2015-04-26 20:36:51,066 ERROR: org.graylog2.periodical.IndexRetentionThread - Could not re-calculate index ranges after running retention: Maximum concurrency of job is reached.
2015-04-26 20:37:02,855 INFO : org.graylog2.indexer.ranges.RebuildIndexRangesJob - Calculated range of [graylog2_2673] in [18637ms].
2015-04-26 20:37:03,038 INFO : org.graylog2.indexer.ranges.RebuildIndexRangesJob - Calculated range of [graylog2_2663] in [182ms].
2015-04-26 20:37:03,200 INFO : org.graylog2.indexer.ranges.RebuildIndexRangesJob - Calculated range of [graylog2_2662] in [162ms].
2015-04-26 20:37:04,283 INFO : org.graylog2.indexer.ranges.RebuildIndexRangesJob - Calculated range of [graylog2_2661] in [1083ms].
2015-04-26 20:37:05,464 INFO : org.graylog2.indexer.ranges.RebuildIndexRangesJob - Calculated range of [graylog2_2670] in [1180ms].
2015-04-26 20:37:07,907 INFO : org.graylog2.indexer.ranges.RebuildIndexRangesJob - Calculated range of [graylog2_2660] in [2442ms].
2015-04-26 20:37:07,923 INFO : org.graylog2.indexer.ranges.RebuildIndexRangesJob - Done calculating index ranges for 21 indices. Took 60852ms.
2015-04-26 20:37:07,924 INFO : org.graylog2.system.jobs.SystemJobManager - SystemJob <dd41b870-ec53-11e4-9fc2-fa163e8ba044> [org.graylog2.indexer.ranges.RebuildIndexRangesJob] finished in 61276ms.
```

**Comment by JeremyEinfeld** [comment]
Graylog version 1.0.1

**Comment by kroepke** [comment]
Thanks, we will try to address this in 1.1 but I cannot promise it yet.

**Comment by bernd** [comment]
@joschi I think this is fixed with the latest index range changes, right?

**Comment by joschi** [comment]
Yes, should be fixed by the index range refactoring in #1274.
